services/core-web/src/components/noticeOfWork/NOWStatusIndicator.js

```diff
@@ -25,55 +28,134 @@ import { getNoticeOfWork, getApplicationDelay } from "@common/selectors/noticeOf
+  delayTypeOptionsHash: PropTypes.objectOf(PropTypes.string).isRequired,
+  tabSection: PropTypes.string,
 };
 
-const defaultProps = { isComplete: "", isEditMode: false };
+const defaultProps = { isEditMode: false, tabSection: "" };
 
-export const NOWStatusIndicator = (props) => {
-  const [color, setColor] = useState(
-    "linear-gradient(90deg, #FDBC00 0%, #FDBC00 28.89%, #FFF1A7 100%)"
-  );
+export class NOWStatusIndicator extends Component {
+  state = { bannerColor: "", badgeColor: "", message: "" };
 
-  const isApplicationDelayed = !isEmpty(props.applicationDelay);
-  const message = "Complete";
-  return (
-    <>
-      {props.type === "banner" && (
-        <Alert
-          showIcon={false}
-          message={message}
-          banner
-          style={{
-            background: color,
-            display: "none",
-          }}
-          className="status-banner"
-        />
-      )}
-      {props.type === "badge" && (
-        <Tooltip title="This will say the status" placement="top" mouseEnterDelay={0.3}>
-          <Badge
-            color={color}
+  componentDidMount() {
+    this.handleIndicatorColor(
+      this.props.applicationDelay,
+      this.props.noticeOfWork.now_application_status_code,
+      this.props.isEditMode,
+      this.props.progress,
+      this.props.tabSection
+    );
+  }
+
+  componentWillReceiveProps(nextProps) {
+    const delayChanged = this.props.applicationDelay !== nextProps.applicationDelay;
+    const statusChanged =
+      this.props.noticeOfWork.now_application_status_code !==
+      nextProps.noticeOfWork.now_application_status_code;
+    const progressChanged = this.props.progress !== nextProps.progress;
+    const editModeChanged = this.props.isEditMode !== nextProps.isEditMode;
+
+    if (delayChanged || statusChanged || progressChanged || editModeChanged) {
+      this.handleIndicatorColor(
+        nextProps.applicationDelay,
+        nextProps.noticeOfWork.now_application_status_code,
+        nextProps.isEditMode,
+        nextProps.progress,
+        nextProps.tabSection
+      );
+    }
+  }
+
+  handleIndicatorColor = (applicationDelay, statusCode, isEditMode, progress, tabSection) => {
+    const isApplicationDelayed = !isEmpty(applicationDelay);
+    const isApplicationComplete =
+      statusCode === "AIA" || statusCode === "WDN" || statusCode === "REJ";
+
+    if (isApplicationComplete) {
+      if (statusCode === "AIA") {
+        this.setState({
+          bannerColor:
+            "linear-gradient(90deg, #45A766 0%, #45A766 28.89%, rgba(127,254,0,0.13) 100%)",
+          badgeColor: "#45A766",
+          message: "Application is Approved",
+        });
+      } else {
+        const message =
+          statusCode === "WDN" ? "Application has been Withdrawn" : "Application has been Rejected";
+        this.setState({
+          bannerColor: "linear-gradient(77.2deg, #D40D0D 0%, #E70000 28.89%, #FFFFFF 100%)",
+          badgeColor: "#D40D0D",
```

Comment:
This colour and others defined in this file exist as constants in the file `styles.js`. Can we use that and also add the `bannerColor` attributes to that as well?

Reply:
@mmumby ?

Reply:
Yes - @exzizt I'll do this change in my PR